examples/clusterclasses/docker/rke2/clusterclass-docker-rke2.yaml

```diff
@@ -71,6 +77,42 @@ spec:
               path: /spec/template/spec/customImage
               valueFrom: 
                 variable: dockerImage
+    - name: dockerAuthSecret
+      definitions:
+        - selector:
+            apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+            kind: RKE2ControlPlaneTemplate
+            matchResources:
+              controlPlane: true
+          jsonPatches:
+            - op: add
+              path: /spec/template/spec/privateRegistriesConfig/configs
+              valueFrom:
+                template: |-
+                  {{- if .dockerAuthSecret }}
+                  "registry-1.docker.io":
+                    authSecret:
+                      name: {{ .dockerAuthSecret }}
+                      namespace: {{ .builtin.cluster.namespace }}
+                  {{- end }}
+        - selector:
+            apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+            kind: RKE2ConfigTemplate
+            matchResources:
+              machineDeploymentClass:
+                names:
+                - default-worker
+          jsonPatches:
+            - op: add
+              path: /spec/template/spec/privateRegistriesConfig/configs
+              valueFrom:
+                template: |-
+                  {{- if .dockerAuthSecret }}
```

Comment:
I think this should be replaced with enabledIf: '{{ ne .dockerAuthSecret "" }}'

Reply:
done

Reply:
@mjura  The e2e tests seem to be failing with this change. Also, these changes might not be necessary since it already works fine in the vSphere example: https://github.com/rancher/turtles/blob/main/examples/clusterclasses/vsphere/rke2/clusterclass-rke2-example.yaml#L362.
Would you be okay if we consider using an if-else approach instead?

Reply:
yes, thank you for testing it, please use `if-else` approach